src/EnergyPlus/EcoRoofManager.cc

```diff
@@ -705,16 +707,17 @@ namespace EcoRoofManager {
         using namespace DataEnvironment;
 
         auto const *thisMat = state.dataMaterial->Material(state.dataConstruction->Construct(ConstrNum).LayerPoint(1));
+        auto const *thisMatChild = dynamic_cast<const Material::MaterialChild *>(thisMat);
```

Comment:
Again, just `auto const *thisMat = dynamic_cast<const Material::MaterialChild *>(state.dataMaterial->Material(state.dataConstruction->Construct(ConstrNum).LayerPoint(1));`  is cleaner and easier to review. If you want to shorten the line, then `auto &thisConstruction` is probably a better idea.

Reply:
will change to this format instead